### DISCUSSION TITLE
A question related to linking jbrowse2

### DISCUSSION THREAD

**Discussion by alslonik** [GMOD/jbrowse-components#4482] [Q&A] [answered]
I would like to present  jbrowse2 linear view of two reference sequences side by side by link on a page of my webdb. I am not sure how to do it via url query parameters or via config.json. I want a simple linear genome view for two orthologous sequences in two species, that I will be able to link to it, from my database website. Is it possible?

**Comment by cmdcolin** [comment, marked as answer]
@alslonik 
there are a couple options for this


in jbrowse 2 it is actually a bit hard to present two different species "side by side"
you can do it as "top and bottom" though relatively easily

you can either

a) use two different linear genome views, showing two different assemblies
b) use a single synteny view, showing the two different assemblies



here is an example of (a) using just two linear genome views


https://jbrowse.org/code/jb2/main/?config=test_data%2Fconfig_demo.json&session=spec-{"views":[{"assembly":"hg19","loc":"17:41,196,294..41,277,363","type":"LinearGenomeView","tracks":["ncbi_gff_hg19"]},{"assembly":"hg38","loc":"17:43,044,294..43,125,363","type":"LinearGenomeView","tracks":["ncbi_refseq_109_hg38_latest"]}]}



here is an example of (b) using the synteny view

https://jbrowse.org/code/jb2/main/?config=test_data%2Fconfig_demo.json&session=spec-{"views":[{"type":"LinearSyntenyView","tracks":["hg19ToHg38.over.chain.gz-1645073157673"],"views":[{"assembly":"hg19","loc":"17:41,196,294..41,277,363","type":"LinearGenomeView","tracks":["ncbi_gff_hg19"]},{"assembly":"hg38","loc":"17:43,044,294..43,125,363","type":"LinearGenomeView","tracks":["ncbi_refseq_109_hg38_latest"]}]}]}

more info on the url prams here
https://jbrowse.org/jb2/docs/urlparams/

you can also in some cases use a gene name in the &loc= paramter if you added text indexing, but hope this helps.

note that these url parameters are not available if you use embedded...if you use our embedded/npm components let me know

**Reply by cmdcolin** [reply]
(also note that for b, the large synteny block is just red because it's in a big exact match :))

**Comment by alslonik** [comment]
Hi Colin,
Thanks!
Actually,  actually I meant "top and bottom", and this is the answer I
needed.
I do not need synteny for this specific presentation, just two genomes and
two orthologous sequences.
So, this is what I needed.
I definitely will use  gene names for loc=.
Yes, I noticed that these url parameters are not available for embedded.
So, I will use a regular web version.
I use Jbrowse1 iframe for a single gene presentation on the site now. Maybe
will use jbrowse2 embedded genome linear view after I migrate the main
browser to Jbrowse2.
Thanks!
Alex



On Thu, 18 Jul 2024 at 15:35, Colin Diesh ***@***.***> wrote:

> (also note that for b, the large synteny block is just red because it's in
> a big exact match :))
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GMOD/jbrowse-components/discussions/4482#discussioncomment-10084254>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJFOT34MKYF4MSMHI6RIAWDZM6ZA7AVCNFSM6AAAAABK77T2GOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMBYGQZDKNA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 


Kind regards,

Alexandra Slonimsky
